rfcs/0000_code_organization_guide.md

```diff
@@ -0,0 +1,555 @@
+```
+
+What if we want to make the area computer configurable?
+
+```python
+# Define the protocol
+class AreaComputer(Protocol):
+	@staticmethod
+	def area(length: float, height: float) -> float: ...
+
+# Update Rectangle to accept area computer
+class Rectangle:
+	def __init__(
+		self,
+		length: float,
+		height: float,
+		area_computer: type[AreaComputer] = TooComplicatedAreaComputer
+	) -> None:
+		super().__init__()
+		self._length = length
+		self._height = height
+		self._area_computer = area_computer
+
+	# --unchanged code omitted--
+```
+
+If we want our code to change rapidly, to try out different ideas, and to configure existing code with these variants, using modular components for functionality reuse instead of inheritance allows for changes to remain small in scope without affecting unrelated functionality up and down the inheritance chain.
+
+## Bare functions or static methods SHOULD be used to share a functionality implementation that does not access instance state
+
+_Why_: Do not require state that you don’t access. Functions without state are vastly easier to reuse, refactor, reason about, and test.
+
+```python
+# calculating an area
+class DefaultAreaComputer:
+	@staticmethod
+	def area(length: float, height: float) -> float:
+		return length * height
+
+# alternatively
+def area(length: float, height: float) -> float:
+	return length * height
+```
+
+A reason to use a static method on a class over a bare function would be when we want to pass the functionality to another class. This is because we want our configurations to be serializable, and a type is serializable in a more straightforward manner than a Callable would be. For example:
+
+```python
+# Static method approach
+class Rectangle:
+	def __init__(
+		self,
+		length: float,
+		height: float,
+		area_computer: type[AreaComputer] = DefaultAreaComputer # Easier to serialize
+	) -> None:
+		super().__init__()
+		self._length = length
+		self._height = height
+		self._area_computer = area_computer
+
+	@property
+	def area(self) -> float:
+		return self._area_computer.area(self._length, self._height)
+
+# Bare function approach
+class Rectangle:
+	def __init__(
+		self,
+		length: float,
+		height: float,
+		area_computer: Callable[[float, float], float] = area # More challenging to serialize
+	) -> None:
+		super().__init__()
+		self._length = length
+		self._height = height
+		self._area_computer = area_computer
+
+	@property
+	def area(self) -> float:
+		return self._area_computer(self._length, self._height)
+```
+
+## To share a functionality implementation that reads the state of the instance being mixed with, mixins MAY be used
+
+For sharing functionality, mixins only implement a shared behaves-like-a functionality. They add functionality, they do not add state. However, mixins SHALL NOT add state to the instance being mixed with. Every time you find yourself in need of state when working on a mixin, switch to composition instead.
```

Comment:
```suggestion
For sharing functionality, mixins only implement a shared behaves-like-a functionality. They add functionality, however, mixins SHALL NOT add state to the instance being mixed with. Every time you find yourself in need of state when working on a mixin, switch to composition instead.
```

Reply:
What's the reason for this suggestion? The intent is that, in the context of allowable mixin use, they add functionality, but they do not add state.

Reply:
That's what the sentence says after the "however". This change looks like an improvement to me.
> They add functionality, however, mixins SHALL NOT add state to the instance being mixed with.

Reply:
Ok, 2 to 1 takes it :)